### ISSUE TITLE
Syntax highlighting breaks when function accepting Char has a trailing tick

### ISSUE THREAD

**Issue by NotTheEconomist** [SublimeHaskell/SublimeHaskell#368]
The following valid Haskell code breaks the poor syntax highlighter (which is trying its best).

    f' :: Char -> Char
    f' = id

    f' '5'  -- 5 here does not appear highlighted as a Char

**Comment by bscottm** [comment]
@NotTheEconomist: Yeah. Syntax highlighting needs to upgrade to Sublime 3's approach, which takes time, in between other issue reports.

But I'm working on it. It's just nowhere near ready for prime time.

**Comment by NotTheEconomist** [comment]
As I'm sure you know there's a number of cases where syntax highlighting breaks, but here's a couple more:

```haskell
value1 = let someExpr = someCalculation
         in someExpr
value2 = somethingElse
-- value2's highlighting is broken
```

```haskell
value1 = someExpr
  where
  someExpr = someCalculation
-- someExpr's highlighting is broken
```

In both cases here, the tokenized word should include the `entity.name.function.haskell` group.

**Comment by bscottm** [comment]
@NotTheEconomist: `f' '5'` is a tough one because ST3 sees a valid Haskell character `' '` followed by something that should be a character. I'll keep jousting at the windmill -- the current (and the ST3-updated) syntax both do not do a good job on function invocations. Yet.
